Library/Formula/postgresql.rb

```diff
@@ -77,8 +82,22 @@ def install
   end
 
   def post_install
-    unless File.exist? "#{var}/postgres"
-      system "#{bin}/initdb", "#{var}/postgres"
+    pg_version = (var/"postgres/PG_VERSION").read.chomp
+
+    if (var/"postgres").exist? && (var/"postgres-#{pg_version}").exist?
+      puts "We have not attempted to migrate your data due to your multi-postgresql setup."
+    end
+
+    # With the 9.5 release, we'll need to adapt logic to make sure we preserve the
+    # 9.4 var directories, without breaking postgres. This should be able to be done
+    # by modifying some lines below to var/"postgres-4.9" instead of var/"postgres"
+    if (var/"postgres").exist?
+      mv var/"postgres", var/"postgres-#{pg_version}"
+      # So people's scripts, etc don't insta-break, symlink this back.
+      ln_sf var/"postgres-#{pg_version}", var/"postgres"
```

Comment:
Although that example would need to be modified to fit the need to warn users in the situation the versioned path is already present.